### PR TITLE
IR-685: Fix questions/answers case and add question marks to questions

### PR DESCRIPTION
**Pull request by xoen**
A lot of the questions/answers in NOMIS is in "SHOUT CASE" and very often questions don't have a question mark.

This PR attempts to solve these problems. Bearing in minds languages are hard and this is automated.
In some cases (no pun intended) the case may not be exactly what a human would want, for example the string `IT` may mean the pronoun "it" or it may refer to "IT" (as in Information Technology).

Key points:
- try to do the right thing as much as possible, acknowledging that sometimes the case may not be perfect (this potentially means a sentence with the perfect case may be translated in something not as good)
- try to preserve the case of some words (e.g. `SIM`, `ACCT`, etc...)
- for questions' labels, add a question mark (?) when it's obvious the sentence is a question (e.g. `WAS THE POLICE INFORMED`, "clearly" this is a question)
- remove duplicated spaces that sometimes are present 
- the questions/answers codes are retained, only the labels (user-facing strings) are converted. This is very important because the codes are used to identify questions/answers

How to review:
- I went through a few iterations of the conversion logic, this may or may not be very interesting. Review commit-by-commit may be useful to have an overall idea...(and I tried to explain some of the tradeoffs in the commit messages)
- ...but perhaps is best to look at the final logic changes and the diffs of the report configs (`.ts` files)
- the `.svg`/`.dot` files are simply derived from the config files so I'd say are less interesting

### CODE REVIEW

**Review by Mjwillis — APPROVED** [review]
Great work!

File: server/utils/utils.ts [hunk 1]
```diff
@@ -166,3 +166,168 @@ export function getComponentString(macroName: string, params = {}): string {
...
+  ]
+
+  const input = str.trim()
+  if (input.length === 0) {
+    return ''
+  }
+
+  // match words or non-words
+  const regex = /(\w+|[^\w\s]+|\s)/g
+  const parts = input.match(regex)
```
Comment (ushkarev) [inline server/utils/utils.ts:308]: `parts` can be null if for some reason the input does not match (e.g. on empty string), in which case i would simply return the input unchanged

File: server/routes/viewReport.ts [hunk 2]
```diff
@@ -66,3 +65,24 @@ export default function viewReport(service: Services): Router {
...
+ */
+function useReportConfigLabels(report: ReportWithDetails, reportConfig: IncidentTypeConfiguration): ReportWithDetails {
+  for (const question of report.questions) {
+    const questionConfig = reportConfig.questions[question.code]
+    if (!questionConfig) {
+      // eslint-disable-next-line no-continue
+      continue
+    }
+
+    question.question = questionConfig?.label ?? question.code
```
Comment (ushkarev) [inline server/routes/viewReport.ts:80]: shouldn't the fallback be `?? question.question` at the end? that's the human readable bit as taken from nomis

File: server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts [hunk 3]
```diff
@@ -171,7 +171,7 @@ const ATTEMPTED_ESCAPE_FROM_CUSTODY: IncidentTypeConfiguration = {
           id: '179034',
           code: 'RULE 43 UNIT (GOAD)',
           active: true,
-          label: 'RULE 43 UNIT (GOAD)',
+          label: 'Rule 43 unit (goad)',
```
Comment (ushkarev) [inline server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts:174]: google says GOAD might is another acronym (= good order and discipline)

File: server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts [hunk 4]
```diff
@@ -4268,14 +4268,14 @@ const ATTEMPTED_ESCAPE_FROM_CUSTODY: IncidentTypeConfiguration = {
       id: '44920',
       active: true,
       code: 'QUOTE THE VANTIVE CASE NUMBER',
-      label: 'QUOTE THE VANTIVE CASE NUMBER',
+      label: 'Quote the vantive case number',
```
Comment (ushkarev) [inline server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts:4271]: technically a company name but i think it's fine to ignore this one

File: server/reportConfiguration/types/DISORDER.ts [hunk 5]
```diff
@@ -639,7 +639,7 @@ const DISORDER: IncidentTypeConfiguration = {
           id: '214747',
           code: 'ORRU ASSISTANCE REQUESTED',
           active: true,
-          label: 'ORRU ASSISTANCE REQUESTED',
+          label: 'Orru assistance requested',
```
Comment (ushkarev) [inline server/reportConfiguration/types/DISORDER.ts:642]: ORRU is another acronym: Operational Response & Resilience Unit

File: server/reportConfiguration/types/FINDS.ts [hunk 6]
```diff
@@ -108,7 +108,7 @@ const FINDS: IncidentTypeConfiguration = {
           id: '218694',
           code: 'HANDHELD METAL DETECTOR (HMD) WAND',
           active: true,
-          label: 'HANDHELD METAL DETECTOR (HMD) WAND',
+          label: 'Handheld metal detector (hmd) wand',
```
Comment (ushkarev) [inline server/reportConfiguration/types/FINDS.ts:111]: maybe HMD should be an acronym

File: server/utils/utils.ts [hunk 7]
```diff
@@ -166,3 +166,168 @@ export function getComponentString(macroName: string, params = {}): string {
...
+    'DYI',
+    'EGS',
+    'F2052SH',
+    'F78A',
+    'FES',
+    'GP',
+    'H.C.C.',
+    'HCO',
+    'HHMD',
+    'HMPS',
```
Comment (ushkarev) [inline server/utils/utils.ts:263]: i think we need HMPPS here too even if it's not used yet

File: server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts (hunk 4 above)
Comment (xoen) [reply server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts:4271]: Is it? I saw this occur several times but when I looked it up (as I didn't know what it meant) and I ended up with this definition:

> Noun (Civil Service jargon). A query for which there is no clear procedure or instruction, and awaits clarification.

And therefore I left it alone, but it's referring to a CRM I'll add to the list of words to preserve as `Vantive` then?

File: server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts (hunk 3 above)
Comment (xoen) [reply server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts:174]: I saw this a few times but I didn't find anything useful. I'll add it as a preserved word as `GOAD` then unless you have objections.

File: server/reportConfiguration/types/DISORDER.ts (hunk 5 above)
Comment (xoen) [reply server/reportConfiguration/types/DISORDER.ts:642]: Didn't sport this, I'll add `ORRU` to the preserved words 👍

File: server/reportConfiguration/types/FINDS.ts (hunk 6 above)
Comment (xoen) [reply server/reportConfiguration/types/FINDS.ts:111]: Ah! I thought I added that one, I'll add with the other words mentioned 👍

File: server/utils/utils.ts (hunk 1 above)
Comment (xoen) [reply server/utils/utils.ts:308]: `input` is trimmed and returned as-is if it's an empty string (well, if it's length is `0`) - so it shouldn't be an issue? 🤔 

Beside mapped would then be an empty array and `.join('')` would turn into an empty string anyway no?

File: server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts (hunk 3 above)
Comment (xoen) [reply server/reportConfiguration/types/ATTEMPTED_ESCAPE_FROM_CUSTODY.ts]: Updated here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/790cdaf0d52e8865428c4d05bd127067f162acdf

File: server/reportConfiguration/types/DISORDER.ts (hunk 5 above)
Comment (xoen) [reply server/reportConfiguration/types/DISORDER.ts]: Updated here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/790cdaf0d52e8865428c4d05bd127067f162acdf

File: server/reportConfiguration/types/FINDS.ts (hunk 6 above)
Comment (xoen) [reply server/reportConfiguration/types/FINDS.ts]: Updated here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/790cdaf0d52e8865428c4d05bd127067f162acdf

File: server/utils/utils.ts (hunk 7 above)
Comment (xoen) [reply server/utils/utils.ts:266]: Updated here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/790cdaf0d52e8865428c4d05bd127067f162acdf

File: server/routes/viewReport.ts (hunk 2 above)
Comment (xoen) [reply server/routes/viewReport.ts:80]: You are correct: The intent was that if for whatever reason the question/answer config couldn't be found (possibly mainly test data with bogus questions/responses) we would leave the question/response as returned from the API.

Comment (xoen) [reply server/routes/viewReport.ts]: Corrected here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/0f991237edffd7e8d58318ff3cbcab8e4cf5efb5

**Review by ushkarev — APPROVED** [review]

**Pull request merged**